File: be_java_elearning/elearning_support_backend/src/main/java/com/elearning/elearning_support/models/mongo/LoginHistoryDocument.java
package com.elearning.elearning_support.models.mongo;

import java.util.Date;
import javax.persistence.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import com.elearning.elearning_support.entities.auth.AuthInfo;
import com.elearning.elearning_support.utils.DateUtils;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document(value = "login_history")
public class LoginHistoryDocument {

    @Id
    private String id;

    private Long userId;

    private String ipAddress;

    private Date loggedInAt;

    public LoginHistoryDocument(AuthInfo authInfo) {
        this.setUserId(authInfo.getUserId());
        this.setIpAddress(authInfo.getIpAddress());
        this.setLoggedInAt(DateUtils.asDate(authInfo.getLastLoginAt()));
    }
}

File: be-gogin/internal/domains/auth/models/mongo/login_history.document.go
package mongo

import (
	"go.mongodb.org/mongo-driver/v2/bson"
	"time"
)

type LoginHistoryDocument struct {
	// _id tự sinh, phải có omitempty nếu không -> zero value 000000....
	Id          bson.ObjectID `bson:"_id,omitempty"`
	UserId      int64         `bson:"userId"`
	LoggedInAt  time.Time     `bson:"loggedInAt"`
	IpAddresses string        `bson:"ipAddress"`
}
